### PR TITLE
r.distance: Add tests for r.distance module

### PR DESCRIPTION
**Pull request by Shreshth-Malik**
Tests added:

Standard Distance: Tests distances between `roadsmajor` and `lakes`.
Overlap : Tests distances within the `lakes` map using the `-o` flag.
Exclude Zero Distance : Tests distances between `lakes` and `roadsmajor` using `-n` flag.

### CODE REVIEW

**Comment by echoix** [comment]
There are test failures on Windows (1 test) https://github.com/OSGeo/grass/actions/runs/11832847802/job/32970371366?pr=4679#step:13:931

File: raster/r.distance/testsuite/test_distance.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            "g.remove", flags="f", type="raster", name=["map1", "map2", "map3"]
+        )
+        cls.del_temp_region()
+
+    def test_distance(self):
+        """Test distance calculation between map1 and map2."""
+        module = SimpleModule("r.distance", map=("map1", "map2"))
+        self.assertModule(module)
+
+        result = module.outputs.stdout.strip().split("\n")
```
Comment (petrasovaa) [inline raster/r.distance/testsuite/test_distance.py:40]: This should use `splitlines()`, see if that fixes the error on Windows

File: raster/r.distance/testsuite/test_distance.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        expected_results = ["1:1:0:0.5:9.5:0.5:9.5"]
+
+        self.assertEqual(
+            result,
+            expected_results,
+            "Mismatch in r.distance output for overlapping features",
+        )
+
+    def test_null_distance(self):
+        """Test r.distance when comparing a map to itself with overlapping features."""
```
Comment (petrasovaa) [inline raster/r.distance/testsuite/test_distance.py:65]: This doc is copy pasted from previous test.

**Review by echoix — APPROVED** [review]

**Comment by echoix** [comment]
I mispressed the update button instead.. but it was still good

**Review by echoix — APPROVED** [review]

**Pull request merged**